foxglove_bridge_base/src/serialization.cpp

```diff
@@ -21,14 +21,20 @@ void to_json(nlohmann::json& j, const Parameter& p) {
   } else if (paramType == ParameterType::PARAMETER_STRING_ARRAY) {
     j["value"] = p.getValue<std::vector<std::string>>();
   } else if (paramType == ParameterType::PARAMETER_NOT_SET) {
-    throw std::runtime_error("Unintialized parameter");
+    // empty value.
   }
 
   j["name"] = p.getName();
 }
 
 void from_json(const nlohmann::json& j, Parameter& p) {
   const auto name = j["name"].get<std::string>();
+
+  if (j.find("value") == j.end()) {
+    p = Parameter(name);  // Value is not set (undefined).
+    return;
+  }
```

Comment:
I think it would be more explicit if the value was set to `null` (nlohmann::detail::value_t::boolean)

Reply:
I don't understand the comment here, `null` and undefined are not the same thing.